### PR TITLE
feat: improve resolve error

### LINKED ISSUE #2033: [Bug]: resolve error printed as debug message
### Reproduction link or steps

I'm testing Rolldown in https://github.com/oxc-project/monitor-oxc/tree/rolldown

```
pnpm install
pnpm run rolldown
```

It's currently producing debug error message for resolve errors. The process also hangs.

### What is expected?

Nicer error messages. No hang.

### What is actually happening?

https://github.com/rolldown/rolldown/blob/4ca169ce74548b647c38cbcda6c2991b77a9d3c7/crates/rolldown/src/types/module_factory.rs#L152-L155

```
 ERROR  Rolldown internal error: Unexpectedly failed to resolve dependencies of /Users/boshen/oxc/monitor-oxc/node_modules/.pnpm/eslint-plugin-import@2.29.1_@typescript-eslint+parser@8.2.0_eslint@9.9.0_jiti@1.21.6_supports_zmswtkamuftpl7jgsbtw22axge/node_modules/eslint-plugin-import/lib/rules/no-unused-modules.js. Got errors [
    (
        RawImportRecord {
            module_request: Rstr(
                "eslint/lib/cli-engine/file-enumerator",
            ),
            kind: Require,
            namespace_ref: SymbolRef {
                owner: 1606,
                symbol: SymbolId(
                    311,
                ),
            },
            module_request_start: 1710,
            meta: ImportRecordMeta(
                0x0,
            ),
        },
        PackagePathNotExported(
            "./lib/cli-engine/file-enumerator",
            "/Users/boshen/oxc/monitor-oxc/node_modules/.pnpm/eslint-plugin-import@2.29.1_@typescript-eslint+parser@8.2.0_eslint@9.9.0_jiti@1.21.6_supports_zmswtkamuftpl7jgsbtw22axge/node_modules/eslint/package.json",
        ),
    ),
```

### System Info

```Shell
None
```

### Any additional comments?

_No response_

### PR DESCRIPTION
**Pull request by shulaoda**
### Description
closes #2033 

image

### CODE REVIEW

**Comment by shulaoda** [comment]
@Boshen cc

**Comment by underfin** [comment]
Could you add the test for this?

File: crates/rolldown/src/types/module_factory.rs [hunk 1]
```diff
@@ -93,8 +93,8 @@ impl<'a> CreateModuleContext<'a> {
   pub async fn resolve_dependencies(
     &mut self,
     dependencies: &IndexVec<ImportRecordIdx, RawImportRecord>,
-  ) -> anyhow::Result<IndexVec<ImportRecordIdx, ResolvedId>> {
-    let jobs = dependencies.iter_enumerated().map(|(idx, item)| {
+  ) -> anyhow::Result<Result<IndexVec<ImportRecordIdx, ResolvedId>, Vec<BuildDiagnostic>>> {
```
Comment (IWANABETHATGUY) [inline crates/rolldown/src/types/module_factory.rs]: https://github.com/rolldown/rolldown/blob/d023f2f95d604a3ab2ce0d627795fc8a3b77bd99/crates/rolldown_error/src/type_aliases.rs#L3-L3, use this, please.

Comment (shulaoda) [reply crates/rolldown/src/types/module_factory.rs]: Yes, I remember this, but I forgot where it was. 😂

**Comment by shulaoda** [comment]
> Could you add the test for this?

Thank you for the reminder!

**Comment by shulaoda** [comment]
Unbelievable, this test case is too torturous! 😭

**Review by IWANABETHATGUY — APPROVED** [review]

**Pull request merged**